src/Compatibility/Microsoft.DotNet.PackageValidation/GetLastStablePackage.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.Build.Framework;
+using Microsoft.NET.Build.Tasks;
+using NuGet.Common;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.PackageValidation
+{
+    public class GetLastStablePackage : TaskBase
+    {
+        [Required]
+        public string PackageId { get; set; }
+
+        public string[] NugetFeeds { get; set; }
+
+        public string LocalPackagesPath { get; set; }
+
+        public bool UseLocalPackagesPath { get; set; }
+
+        [Output]
+        public string LastStableVersion { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            NuGetVersion version = null;
+            if (UseLocalPackagesPath)
+            {
+                IEnumerable<NuGetVersion> versions = null;
+                versions = GetLatestPackageFromLocalFeed(LocalPackagesPath);
+                NuGetVersion packageVersion = versions?.Where(t => !t.IsPrerelease).OrderByDescending(t => t.Version).FirstOrDefault();
+                LastStableVersion = packageVersion?.ToNormalizedString();
+                return;
+            }
+
+            foreach (string nugetFeed in NugetFeeds)
```

Comment:
If multiple feeds are passed in, the latest one which contains the package always wins. Is that what you want?

Reply:
getting the latest package from all the feeds

Reply:
The current logic isn't doing that. The latest feed always wins.

Reply:
The last version will only win if it has the greater package version than the previous feeds here https://github.com/dotnet/sdk/pull/16992/files/15d8c44ca003bd5a9495a13436bfc04c94f8c82f..ae4d1fc5c98621fd3c35197cd8046fb2db50b480#diff-b8dc682a4ab2079ab3cf51ea7e65c2288ba4c0db3ab1cd4ab84e3162e1ce2cecR59

Reply:
Understood it now, thanks for the clarification.